### PR TITLE
{{dev readme requisites 20190917}}

### PR DESCRIPTION
**Pull request by geethaaluri**
Data preprocessing.py has the functions for cleaning data

Ratings_matrix.py has the rating matrix and sparsity calculations

Recommendations_cf has the function for top_n recommendations

matrix_factorization_gridsearch has the grid search function for svd, nmf

**model_cf.py is the main file which uses the built functions**

### CODE REVIEW

**Comment by barisozmen** [comment]
@geethaaluri , Could you rename the branch, _{{dev readme requisites 20190917}}_, to a more understandable and simple phrase/word?

File: Insight_Project_Framework/data_preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+# data = pd.read_csv('data_subset.csv')
+#
+# # data formatted to be able to use surprise package
+# data_surprise = data[['customer_id', 'product_id','star_rating']].\
+#     rename(columns={'customer_id': 'userID', 'product_id': 'itemID', 'star_rating': 'rating'})
+#
+# data_surprise.to_csv('data_surprise.csv', index=False)
+
+def get_data(_file_path,_save_path):
```
Comment (barisozmen) [inline Insight_Project_Framework/data_preprocessing.py:14]: What is the reason using underscores as prefix to parameter names?

**Review by barisozmen — COMMENTED** [review]
I like one-line summaries about each file added/edited by the pull request. 

Could you also write a one-two line summary for what is this branch doing overall? Which functionalities does it add? Is it runnable?

File: Insight_Project_Framework/data_preprocessing.py (hunk 1 above)
Comment (geethaaluri) [reply Insight_Project_Framework/data_preprocessing.py:14]: The underscores in general are to differentiate class/function variables with local variables. Not needed in this case I think.

Comment (geethaaluri) [reply Insight_Project_Framework/data_preprocessing.py:14]: Renaming the branch deleted it and added the same branch with a new name. Sorry, I am new to Git and I didnt know.

Let me submit another pull request basically doing the same.

**Pull request closed without merging**